src/main/java/seedu/planner/model/ModulePlanner.java

```diff
@@ -305,11 +321,57 @@ public void suggestModules(int index) {
         List<Module> modulesTakenBeforeIndex = getAllModulesTakenBeforeIndex(index);
         List<Module> allModules = getAllModulesFromStorage();
 
-        for (Module m: allModules) {
+        // Step 1. Filter modules that user can actually take.
+        for (Module m : allModules) {
             if (ModuleUtil.isModuleAvailableToTake(modulesTaken, modulesTakenBeforeIndex, m)) {
                 modulesAvailable.add(m);
             }
         }
+
+        Map<Major, MajorDescription> map;
+        try {
+            URL resource = MainApp.class.getResource("/data/majorDescription.json");
+            String text = Resources.toString(resource, Charsets.UTF_8);
+            map = JsonUtil.getObjectMapper().readValue(text, MajorDescription.MAP_TYPE_REF);
+        } catch (IOException e) {
+            logger.warning("Unable to read majorDescription file. Start with an empty map.");
+            map = new HashMap<>();
+        }
+
+        // Note: Collections.sort uses stable sort when sorting objects, which we are exploiting here so that
+        // we can chain our sorting and still making sure that the order created by each comparator is preserved.
+        //
+        // The order of comparators you applied to the list matters!
+
+        if (map.containsKey(userProfile.getMajor())) {
```

Comment:
Might want to move some parts into their own methods

Reply:
Done.